### PR TITLE
Connecting redux

### PR DESCRIPTION
**Pull request by DmitriyKrivovyaz**

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -25,6 +27,21 @@ function App() {
     setTheme(newTheme);
   }
 
+  const initializeApp = async () => {
+    await getAndSetAccessToken();
+    await setCurrentLocation();
```
Comment (anton-vlasik) [inline client/src/App.jsx]: These need to move to actions

File: client/src/AppContainer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { getLocationAction } from './redux/actions/locationAction';
+import { getWeatherAction } from './redux/actions/currentWeatherActions';
+import { getHourlyWeatherAction } from './redux/actions/hourlyWeatherAction';
+import { getDailyWeatherAction } from './redux/actions/dailyWeatherActions';
+
+const mapStateToProps = (state, ownProps) => ({
+  location: state.location.location
+});
+
+const getWeather = (id, dispatch) => {
```
Comment (anton-vlasik) [inline client/src/AppContainer.js]: All that need to go to actions

File: client/src/AppContainer.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  dispatch(getWeatherAction(id));
+  dispatch(getHourlyWeatherAction(id));
+  dispatch(getDailyWeatherAction(id));
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  return {
+    getLocation: location => dispatch(getLocationAction(location)),
+    getWeather: id => getWeather(id, dispatch)
```
Comment (anton-vlasik) [inline client/src/AppContainer.js]: no need to pass dispatch here

File: client/src/redux/actions/dailyWeatherActions.js [hunk 4]
```diff
@@ -1,3 +1,16 @@
...
+export const setDailyWeatherAction = dailyWeather => ({
+  type: SET_DAYS,
+  payload: {
+    dailyWeather
+  }
+});
+
+export const getDailyWeatherAction = id => dispatch => {
+  getDailyWeather(id).then(dailyWeather => {
+    console.log(dailyWeather);
```
Comment (anton-vlasik) [inline client/src/redux/actions/dailyWeatherActions.js]: remove console.log

File: client/src/api/weatherApi.js [hunk 5]
```diff
@@ -1,6 +1,11 @@
 import { proxyServer, weatherApi } from './index';
 import { urls } from '../globalConsts';
 
+export const getAndSetAccessToken = async () => {
+  const token = await getAccessToken();
```
Comment (anton-vlasik) [inline client/src/api/weatherApi.js]: Is there anywhere at least one try catch block. ?

**Review by ximet — APPROVED** [review]

**Review by anton-vlasik — APPROVED** [review]

**Pull request merged**